### ISSUE TITLE
JobUpdater crashing to update workqueue elements

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#9562]
**Impact of the bug**
WMAgent, JobUpdater crashes

**Describe the bug**
From the traceback, it looks like JobUpdater is trying to update workqueue elements (which database? local or remote?) that no longer exist, thus failing with the `TypeError`.

**How to reproduce it**
not sure

**Expected behavior**
The component should not crash in a case like that. It should log the error and move on.
In addition to that, I think it would be useful to make the component log slightly more verbose, at least logging which workflow is getting re-prioritized in the component.

**Additional context and error message**
Traceback from the component log in vocms0283:
```
2020-03-18 08:44:38,751:139714867345152:INFO:JobUpdaterPoller:Synchronizing priorities with ReqMgr...
2020-03-18 08:44:39,184:139714867345152:INFO:JobUpdaterPoller:Found 1 workflows to update in workqueue
2020-03-18 08:44:39,193:139714867345152:ERROR:JobUpdaterPoller:Caught unexpected exception in JobUpdater: 'NoneType' object is not iterable
Traceback (most recent call last):
  File "/data/srv/wmagent/v1.2.8/sw/slc7_amd64_gcc630/cms/wmagent/1.2.8/lib/python2.7/site-packages/WMComponent/JobUpdater/JobUpdaterPoller.py", line 84, in algorithm
    self.synchronizeJobPriority()
  File "/data/srv/wmagent/v1.2.8/sw/slc7_amd64_gcc630/cms/wmagent/1.2.8/lib/python2.7/site-packages/WMComponent/JobUpdater/JobUpdaterPoller.py", line 138, in synchronizeJobPriority
    self.workqueue.updatePriority(workflow, workflowsToUpdate[workflow])
  File "/data/srv/wmagent/v1.2.8/sw/slc7_amd64_gcc630/cms/wmagent/1.2.8/lib/python2.7/site-packages/WMCore/Services/WorkQueue/WorkQueue.py", line 230, in updatePriority
    self.updateElements(*elementsToUpdate, Priority=priority)
  File "/data/srv/wmagent/v1.2.8/sw/slc7_amd64_gcc630/cms/wmagent/1.2.8/lib/python2.7/site-packages/WMCore/Services/WorkQueue/WorkQueue.py", line 196, in updateElements
    conflictIDs = self.db.updateBulkDocumentsWithConflictHandle(elementIds, eleParams, maxConflictLimit=20)
  File "/data/srv/wmagent/v1.2.8/sw/slc7_amd64_gcc630/cms/wmagent/1.2.8/lib/python2.7/site-packages/WMCore/Database/CMSCouch.py", line 385, in updateBulkDocumentsWithConflictHandle
    conflictDocIDs = self.updateBulkDocuments(doc_ids, updateParams, updateLimits)
  File "/data/srv/wmagent/v1.2.8/sw/slc7_amd64_gcc630/cms/wmagent/1.2.8/lib/python2.7/site-packages/WMCore/Database/CMSCouch.py", line 366, in updateBulkDocuments
    doc.update(j['doc'])
TypeError: 'NoneType' object is not iterable
2020-03-18 08:44:39,246:139714867345152:ERROR:BaseWorkerThread:Error in worker algorithm (1):
Backtrace:
  <WMComponent.JobUpdater.JobUpdaterPoller.JobUpdaterPoller object at 0x7f11efd71dd0> <@========== WMException Start ==========@>
Exception Class: JobUpdaterException
Message: Caught unexpected exception in JobUpdater: 'NoneType' object is not iterable

        ModuleName : WMComponent.JobUpdater.JobUpdaterPoller
        MethodName : algorithm
        ClassInstance : None
        FileName : /data/srv/wmagent/v1.2.8/sw/slc7_amd64_gcc630/cms/wmagent/1.2.8/lib/python2.7/site-packages/WMComponent/JobUpdater/JobUpdaterPoller.py
        ClassName : None
        LineNumber : 110
        ErrorNr : 0

Traceback: 
  File "/data/srv/wmagent/v1.2.8/sw/slc7_amd64_gcc630/cms/wmagent/1.2.8/lib/python2.7/site-packages/WMComponent/JobUpdater/JobUpdaterPoller.py", line 84, in algorithm
    self.synchronizeJobPriority()

  File "/data/srv/wmagent/v1.2.8/sw/slc7_amd64_gcc630/cms/wmagent/1.2.8/lib/python2.7/site-packages/WMComponent/JobUpdater/JobUpdaterPoller.py", line 138, in synchronizeJobPriority
    self.workqueue.updatePriority(workflow, workflowsToUpdate[workflow])

  File "/data/srv/wmagent/v1.2.8/sw/slc7_amd64_gcc630/cms/wmagent/1.2.8/lib/python2.7/site-packages/WMCore/Services/WorkQueue/WorkQueue.py", line 230, in updatePriority
    self.updateElements(*elementsToUpdate, Priority=priority)

  File "/data/srv/wmagent/v1.2.8/sw/slc7_amd64_gcc630/cms/wmagent/1.2.8/lib/python2.7/site-packages/WMCore/Services/WorkQueue/WorkQueue.py", line 196, in updateElements
    conflictIDs = self.db.updateBulkDocumentsWithConflictHandle(elementIds, eleParams, maxConflictLimit=20)

  File "/data/srv/wmagent/v1.2.8/sw/slc7_amd64_gcc630/cms/wmagent/1.2.8/lib/python2.7/site-packages/WMCore/Database/CMSCouch.py", line 385, in updateBulkDocumentsWithConflictHandle
    conflictDocIDs = self.updateBulkDocuments(doc_ids, updateParams, updateLimits)

  File "/data/srv/wmagent/v1.2.8/sw/slc7_amd64_gcc630/cms/wmagent/1.2.8/lib/python2.7/site-packages/WMCore/Database/CMSCouch.py", line 366, in updateBulkDocuments
    doc.update(j['doc'])

<@---------- WMException End ----------@>  File "/data/srv/wmagent/v1.2.8/sw/slc7_amd64_gcc630/cms/wmagent/1.2.8/lib/python2.7/site-packages/WMCore/WorkerThreads/BaseWorkerThread.py", line 182, in __call__
    tSpent, results, _ = algorithmWithDBExceptionHandler(parameters)
  File "/data/srv/wmagent/v1.2.8/sw/slc7_amd64_gcc630/cms/wmagent/1.2.8/lib/python2.7/site-packages/WMCore/Database/DBExceptionHandler.py", line 39, in wrapper
    return f(*args, **kwargs)
  File "/data/srv/wmagent/v1.2.8/sw/slc7_amd64_gcc630/cms/wmagent/1.2.8/lib/python2.7/site-packages/Utils/Timers.py", line 24, in wrapper
    res = func(*arg, **kw)
  File "/data/srv/wmagent/v1.2.8/sw/slc7_amd64_gcc630/cms/wmagent/1.2.8/lib/python2.7/site-packages/WMComponent/JobUpdater/JobUpdaterPoller.py", line 110, in algorithm
    raise JobUpdaterException(msg)

2020-03-18 08:44:39,246:139714867345152:INFO:Harness:>>>Terminating worker threads
2020-03-18 08:44:39,246:139714867345152:ERROR:BaseWorkerThread:Error in event loop (2): <WMComponent.JobUpdater.JobUpdaterPoller.JobUpdaterPoller object at 0x7f11efd71dd0> <@========== WMException Start ==========@>
```

**Comment by amaltaro** [comment]
So, it's a local workqueue database update and, it was crashing right after every restart of the component.
I noticed the local workqueue views were around 19GB (compared to the 2GB of database size). I then stopped everything (agent + services) and removed the workqueue views:
```
cmst1@vocms0283:/data/srv/wmagent/current/install/couchdb/database $ rm -rf .workqueue_design/mrview/2f77345c5cf5065ec0f98a614222c8b2.view 
```
restarted everything and it's no longer crashing.

Still, I think it's a good idea to make it a bit more verbose (no need to backport anything though)
